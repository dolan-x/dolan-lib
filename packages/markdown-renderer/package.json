{
  "name": "@dolan-x/markdown-renderer",
  "version": "0.0.46",
  "author": "Ray <i@mk1.io> (https://raycoder.me)",
  "description": "Shared utils for Dolan",
  "keywords": [
    "dolan",
    "markdown",
    "markdown-renderer",
    "remark"
  ],
  "homepage": "https://github.com/dolan-x/dolan-lib/tree/main/packages/markdown-renderer#readme",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/dolan-x/dolan-lib.git",
    "directory": "/"
  },
  "bugs": {
    "url": "https://github.com/dolan-x/dolan-lib/issues"
  },
  "license": "MIT",
  "sideEffects": false,
  "exports": {
    ".": {
      "types": "./dist/index.d.ts",
      "require": "./dist/index.cjs",
      "import": "./dist/index.mjs"
    }
  },
  "main": "dist/index.cjs",
  "module": "dist/index.mjs",
  "types": "dist/index.d.ts",
  "files": [
    "dist"
  ],
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "unbuild",
    "stub": "unbuild --stub"
  },
  "dependencies": {
    "rehype-katex": "^6.0.2",
    "rehype-stringify": "^9.0.3",
    "remark-breaks": "^3.0.2",
    "remark-gfm": "^3.0.1",
    "remark-html": "^15.0.1",
    "remark-math": "^5.1.1",
    "remark-parse": "^10.0.1",
    "remark-rehype": "^10.1.0",
    "unified": "^10.1.2",
    "unist-util-visit": "^4.1.1"
  }
}
