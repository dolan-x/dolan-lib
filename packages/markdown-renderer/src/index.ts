import { unified } from "unified";
import remarkGfm from "remark-gfm";
import remarkParse from "remark-parse";
import remarkBreaks from "remark-breaks";
import remarkMath from "remark-math";
import remarkRehype from "remark-rehype";
import rehypeKatex from "rehype-katex";
import rehypeStringify from "rehype-stringify";

export const renderer = unified()
  .use(remarkParse)
  .use(remarkBreaks)
  .use(remarkGfm)
  .use(remarkMath)
  .use(remarkRehype)
  .use(rehypeKatex)
  .use(rehypeStringify);

export async function renderMarkdown (md: string) {
  return String(await renderer.process(md));
}
