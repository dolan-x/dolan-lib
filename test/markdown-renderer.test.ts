import { describe, expect, it } from "vitest";
import { renderMarkdown } from "@dolan-x/markdown-renderer";
import { gfm } from "./fixtures/markdown-renderer";

describe("markdown-renderer", () => {
  it("works", async () => {
    expect(await renderMarkdown("# Hello")).toMatchInlineSnapshot("\"<h1>Hello</h1>\"");
  });
  it("should render gfm", async () => {
    expect(await renderMarkdown(gfm)).toMatchSnapshot("markdown-renderer-gfm");
  });
});
