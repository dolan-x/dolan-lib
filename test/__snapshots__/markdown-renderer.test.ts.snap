// Vitest Snapshot v1

exports[`markdown-renderer > should render gfm > markdown-renderer-gfm 1`] = `
"<h1>GFM</h1>
<h2>Autolink literals</h2>
<p><a href=\\"http://www.example.com\\">www.example.com</a>, <a href=\\"https://example.com\\">https://example.com</a>, and <a href=\\"mailto:contact@example.com\\">contact@example.com</a>.</p>
<h2>Footnote</h2>
<p>A note<sup><a href=\\"#user-content-fn-1\\" id=\\"user-content-fnref-1\\" data-footnote-ref aria-describedby=\\"footnote-label\\">1</a></sup></p>
<h2>Strikethrough</h2>
<p><del>one</del> or <del>two</del> tildes.</p>
<h2>Table</h2>
<table>
<thead>
<tr>
<th>a</th>
<th align=\\"left\\">b</th>
<th align=\\"right\\">c</th>
<th align=\\"center\\">d</th>
</tr>
</thead>
</table>
<h2>Tasklist</h2>
<ul class=\\"contains-task-list\\">
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" disabled> to do</li>
<li class=\\"task-list-item\\"><input type=\\"checkbox\\" checked disabled> done</li>
</ul>
<section data-footnotes class=\\"footnotes\\"><h2 id=\\"footnote-label\\" class=\\"sr-only\\">Footnotes</h2>
<ol>
<li id=\\"user-content-fn-1\\">
<p>Big note. <a href=\\"#user-content-fnref-1\\" data-footnote-backref class=\\"data-footnote-backref\\" aria-label=\\"Back to content\\">â†©</a></p>
</li>
</ol>
</section>"
`;

exports[`markdown-renderer > should render github > markdown-renderer-github 1`] = `
"<p>Some references:</p>
<ul>
<li>Commit: <a href=\\"https://github.com/dolan-x/dolan-lib/commit/f8083175fe890cbf14f41d0a06e7aa35d4989587\\"><code>f808317</code></a></li>
<li>Commit (fork): <a href=\\"https://github.com/foo/dolan-lib/commit/f8083175fe890cbf14f41d0a06e7aa35d4989587\\">foo@<code>f808317</code></a></li>
<li>Commit (repo): <a href=\\"https://github.com/remarkjs/remark/commit/e1aa9f6c02de18b9459b7d269712bcb50183ce89\\">remarkjs/remark@<code>e1aa9f6</code></a></li>
<li>Issue or PR (<code>#</code>): <a href=\\"https://github.com/dolan-x/dolan-lib/issues/1\\">#1</a></li>
<li>Issue or PR (<code>GH-</code>): <a href=\\"https://github.com/dolan-x/dolan-lib/issues/1\\">GH-1</a></li>
<li>Issue or PR (fork): <a href=\\"https://github.com/foo/dolan-lib/issues/1\\">foo#1</a></li>
<li>Issue or PR (project): <a href=\\"https://github.com/remarkjs/remark/issues/1\\">remarkjs/remark#1</a></li>
<li>Mention: <a href=\\"https://github.com/wooorm\\"><strong>@wooorm</strong></a></li>
</ul>
<p>Some links:</p>
<ul>
<li>Commit: <a href=\\"https://github.com/remarkjs/remark/commit/e1aa9f6c02de18b9459b7d269712bcb50183ce89\\">remarkjs/remark@<code>e1aa9f6</code></a></li>
<li>Commit comment: <a href=\\"https://github.com/remarkjs/remark/commit/ac63bc3abacf14cf08ca5e2d8f1f8e88a7b9015c#commitcomment-16372693\\">remarkjs/remark@<code>ac63bc3</code> (comment)</a></li>
<li>Issue or PR: <a href=\\"https://github.com/remarkjs/remark/issues/182\\">remarkjs/remark#182</a></li>
<li>Issue or PR comment: <a href=\\"https://github.com/remarkjs/remark-github/issues/3#issue-151160339\\">remarkjs/remark-github#3 (comment)</a></li>
<li>Mention: <a href=\\"https://github.com/ben-eb\\">https://github.com/ben-eb</a></li>
</ul>"
`;
